micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jpa/HibernateMetrics.java

```diff
@@ -178,15 +178,28 @@ public void bindTo(MeterRegistry registry) {
             Statistics::getCloseStatementCount, "status", "closed");
 
         // Second Level Caching
-        Arrays.stream(statistics.getSecondLevelCacheRegionNames())
-                .forEach(regionName ->  {
+
+        long numRegions = Arrays.stream(statistics.getSecondLevelCacheRegionNames())
+                .filter(this::hasDomainDataRegionStatistics)
+                .peek(regionName -> {
                     counter(registry, "hibernate.second.level.cache.requests", "The number of cacheable entities/collections successfully retrieved from the cache",
-                            stats -> stats.getDomainDataRegionStatistics(regionName).getHitCount(), "region", regionName, "result", "hit");
+                            stats -> stats.getSecondLevelCacheStatistics(regionName).getHitCount(), "region", regionName, "result", "hit");
                     counter(registry, "hibernate.second.level.cache.requests", "The number of cacheable entities/collections not found in the cache and loaded from the database",
-                            stats -> stats.getDomainDataRegionStatistics(regionName).getMissCount(), "region", regionName, "result", "miss");
+                            stats -> stats.getSecondLevelCacheStatistics(regionName).getMissCount(), "region", regionName, "result", "miss");
                     counter(registry, "hibernate.second.level.cache.puts", "The number of cacheable entities/collections put in the cache",
-                            stats -> stats.getDomainDataRegionStatistics(regionName).getPutCount(), "region", regionName);
-                });
+                            stats -> stats.getSecondLevelCacheStatistics(regionName).getPutCount(), "region", regionName);
+                })
+                .count();
+
+        if ( numRegions == 0 ) {
+            // Fallback to previous behavior, with the "all" region tag
```

Comment:
I was on the fence, so I went to the "let's fall back to whatever less fancy thing we were doing before". ;)
I am happy to ditch it entirely.

Reply:
But I'll ask Sanne about it.. because it looks like the path that is incrementing should tip over, too.. so it should all work, or should all be bad..